pkg/cosign/kubernetes/webhook/validator.go

```diff
@@ -104,20 +108,23 @@ func (v *Validator) ValidateCronJob(ctx context.Context, c *duckv1.CronJob) *api
 }
 
 func (v *Validator) validatePodSpec(ctx context.Context, ps *corev1.PodSpec, opt k8schain.Options) (errs *apis.FieldError) {
+	// Read the CRD
 	kc, err := k8schain.New(ctx, v.client, opt)
 	if err != nil {
 		logging.FromContext(ctx).Warnf("Unable to build k8schain: %v", err)
 		return apis.ErrGeneric(err.Error(), apis.CurrentField)
 	}
-
-	s, err := v.lister.Secrets(system.Namespace()).Get(v.secretName)
-	if err != nil {
-		return apis.ErrGeneric(err.Error(), apis.CurrentField)
-	}
-
-	keys, kerr := getKeys(ctx, s.Data)
-	if kerr != nil {
-		return kerr
+	var clusterPolicy = schema.GroupVersionResource{Group: "sigstore.dev", Version: "v1alpha1", Resource: "clusterimagepolicies"}
+	
+	imagePolicy, _ := v.dynamicClient.Resource(clusterPolicy).Get(ctx, "image-policy", metav1.GetOptions{})
```

Comment:
This is an important error to not ignore

Reply:
This also hits the API server on every webhook request, where the previous version hit the lister cache on requests.

Reply:
I was also not expecting the use of a singleton here.  How are users expected to know this?  Are there other policy bits in Kubernetes that use a CRD for a global singleton configuration?